### ISSUE TITLE
Properties with explicitly required values throws linting issues

### ISSUE THREAD

**Issue by ChrisMBarr** [darraghoriordan/eslint-plugin-nestjs-typed#185]
I have certain properties that require explicit values to always be sent, like this:

```ts
@IsNotEmpty()
@IsString()
@Equals('ZCATS') //we require the front end to always send us this exact literal string for some reason ¯\_(ツ)_/¯
Profile!: 'ZCATS';

@IsNotEmpty()
@IsBoolean()
@Equals(true) //we require the front end to always send us a literal true boolean for some reason ¯\_(ツ)_/¯
ReleaseData!: true;
```

But I am getting this linting error for both of these cases
```
A non-primitive property with validation should probably use a @Type decorator. If this is an enum use @IsEnum(). If this is a plain object use @IsObject().eslint[@darraghor/nestjs-typed/validated-non-primitive-property-needs-type-decorator](https://github.com/darraghoriordan/eslint-plugin-nestjs-typed/blob/main/src/docs/rules/validated-non-primitive-property-needs-type-decorator.md)
```

I feel like I am validating these correctly for what I need, but this rule is telling me to do something that I'm pretty sure I'm already doing

**Comment by ChrisMBarr** [comment]
Oh wow, well that's great!  Perhaps another workaround would be that if you use the `@Equals('foo')` decorator then no others should be required - if you only care about specific values then the overall type doesn't really matter.

Would your PR also take care of #187 I opened yesterday as well?

**Comment by abrindam** [comment]
I think it will, I definitely fixed union types of literal strings, so I think union types of literal numbers is probably fixed as well!
